routing/index_graph.cpp

```diff
@@ -103,26 +105,25 @@ void IndexGraph::GetEdgeList(Segment const & segment, bool isOutgoing, vector<Se
   RoadPoint const roadPoint = segment.GetRoadPoint(isOutgoing);
   Joint::Id const jointId = m_roadIndex.GetJointId(roadPoint);
 
+  vector<SegmentEdge> rawEdges;
   if (jointId != Joint::kInvalidId)
   {
     m_jointIndex.ForEachPoint(jointId, [&](RoadPoint const & rp) {
-      GetNeighboringEdges(segment, rp, isOutgoing, edges);
+      GetNeighboringEdges(segment, rp, isOutgoing, rawEdges);
     });
   }
   else
   {
-    GetNeighboringEdges(segment, roadPoint, isOutgoing, edges);
+    GetNeighboringEdges(segment, roadPoint, isOutgoing, rawEdges);
   }
 
   // Removing some edges according to restriction rules.
-  vector<SegmentEdge> filteredEdges;
-  filteredEdges.reserve(edges.size());
-  for (SegmentEdge const & e : edges)
+  edges.reserve(rawEdges.size());
+  for (SegmentEdge const & e : rawEdges)
   {
-    if (!IsRestricted(m_restrictions, edges, segment, e.GetTarget(), isOutgoing))
-      filteredEdges.push_back(e);
+    if (!IsRestricted(m_restrictions, rawEdges, segment, e.GetTarget(), isOutgoing))
```

Comment:
Everything's ok now. Only restriction of type No is used at the time of building route.